lib/Features/Common/Views/homepage.dart

```diff
@@ -1,49 +1,172 @@
+    )
+  ];
+
+  @override
+  void initState() {
+    super.initState();
+    _textController = TextEditingController();
+  }
 
-  void _incrementCounter() {
-    setState(() {
-      _counter++;
-    });
+  @override
+  void dispose() {
+    _textController.dispose();
+    super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
-        title: Text(widget.title),
+    return CupertinoPageScaffold(
+      child: Column(
+        children: [
+          SizedBox(
+            height: MediaQuery.of(context).size.height * 0.8,
+            child: CustomScrollView(
+              slivers: <Widget>[
+                const CupertinoSliverNavigationBar(
+                  backgroundColor: CupertinoColors.white,
+                  largeTitle: Text(
+                    "Your Bookmark",
+                    style: TextStyle(
+                      fontSize: 20.0,
+                      fontWeight: FontWeight.w600,
+                    ),
+                  ),
+                ),
+                SliverToBoxAdapter(
+                  child: Container(
+                    margin: const EdgeInsets.symmetric(vertical: 16.0),
+                    padding: const EdgeInsets.symmetric(
+                        horizontal: 16.0, vertical: 8.0),
+                    child: Column(
+                      children: [
+                        const TopicSection(),
+                        const SizedBox(height: 32.0),
+                        Column(
+                          children: [
+                            BookmarkCard(
+                              topic: bookmarks[0].topic,
+                              title: bookmarks[0].title,
+                              thumbnail: bookmarks[0].thumbnail,
+                              createdAt: bookmarks[0].createdAt,
+                              readingLength: bookmarks[0].readingLength,
+                              onPressed: () {},
+                            ),
+                            BookmarkCard(
+                              topic: bookmarks[1].topic,
+                              title: bookmarks[1].title,
+                              thumbnail: bookmarks[1].thumbnail,
+                              createdAt: bookmarks[1].createdAt,
+                              readingLength: bookmarks[1].readingLength,
+                              onPressed: () {},
+                            ),
+                            BookmarkCard(
+                              topic: bookmarks[2].topic,
+                              title: bookmarks[2].title,
+                              thumbnail: bookmarks[2].thumbnail,
+                              createdAt: bookmarks[2].createdAt,
+                              readingLength: bookmarks[2].readingLength,
+                              onPressed: () {},
+                            ),
+                          ],
```

Comment:
ini bisa pake map,
onPressednya nanti bikin function sendiri aja pake parameter topic / title sesuai kebutuhan
```suggestion
                          children: bookmarks.map( 
                          (bookmark) => BookmarkCard( 
                          topic: bookmark.topic,
                          title: bookmark.title,
                          thumbnail: bookmark.thumbnail,
                          createdAt: bookmark.createdAt,
                          readingLength: bookmark.readingLength,
                          onPressed: () {}
                          )
                          ).toList()
```